File: backend/app/schemas/note.py
"""
Pydantic schemas for caregiver note endpoints
"""

from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime
from typing import Optional
from uuid import UUID
from enum import Enum


# ===== ENUMS =====

class NoteCategory(str, Enum):
    """Note category types"""
    MEDICAL = "medical"
    BEHAVIORAL = "behavioral"
    PREFERENCES = "preferences"
    ROUTINE = "routine"
    SAFETY = "safety"
    FAMILY = "family"
    OTHER = "other"


class NotePriority(str, Enum):
    """Note priority levels"""
    NORMAL = "normal"
    IMPORTANT = "important"


# ===== NOTE SCHEMAS =====

class NoteBase(BaseModel):
    """Base schema for caregiver note"""
    title: str = Field(..., min_length=1, max_length=200, description="Note title")
    content: str = Field(..., min_length=1, description="Note content/description")
    category: NoteCategory = Field(..., description="Note category")
    priority: NotePriority = Field(default=NotePriority.NORMAL, description="Note priority")


class NoteCreate(NoteBase):
    """Schema for creating a note"""
    pass


class NoteUpdate(BaseModel):
    """Schema for updating a note"""
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    content: Optional[str] = Field(None, min_length=1)
    category: Optional[NoteCategory] = None
    priority: Optional[NotePriority] = None


class NoteResponse(NoteBase):
    """Schema for note response"""
    id: UUID
    patient_id: UUID
    caregiver_id: Optional[UUID]
    created_at: datetime
    updated_at: datetime

    # Optional: Include caregiver name if needed
    author_name: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class NoteListResponse(BaseModel):
    """Schema for paginated note list response"""
    notes: list[NoteResponse]
    total: int
    page: int = 1
    page_size: int = 50

    model_config = ConfigDict(from_attributes=True)

File: caregiver-dashboard/src/types/note.ts
export enum NoteCategory {
  MEDICAL = 'medical',
  BEHAVIORAL = 'behavioral',
  PREFERENCES = 'preferences',
  ROUTINE = 'routine',
  SAFETY = 'safety',
  FAMILY = 'family',
  OTHER = 'other',
}

export enum NotePriority {
  NORMAL = 'normal',
  IMPORTANT = 'important',
}

export interface Note {
  id: string;
  patient_id: string;
  title: string;
  content: string;
  category: NoteCategory;
  priority: NotePriority;
  created_at: string;
  updated_at: string;
  created_by: string;
  author_name?: string;
}

export interface CreateNoteData {
  title: string;
  content: string;
  category: NoteCategory;
  priority: NotePriority;
}

export interface UpdateNoteData {
  title?: string;
  content?: string;
  category?: NoteCategory;
  priority?: NotePriority;
}

export const NOTE_CATEGORY_LABELS: Record<NoteCategory, string> = {
  [NoteCategory.MEDICAL]: 'Medical',
  [NoteCategory.BEHAVIORAL]: 'Behavioral',
  [NoteCategory.PREFERENCES]: 'Preferences',
  [NoteCategory.ROUTINE]: 'Routine',
  [NoteCategory.SAFETY]: 'Safety',
  [NoteCategory.FAMILY]: 'Family',
  [NoteCategory.OTHER]: 'Other',
};

export const NOTE_CATEGORY_COLORS: Record<NoteCategory, string> = {
  [NoteCategory.MEDICAL]: 'bg-red-100 text-red-700 border-red-200',
  [NoteCategory.BEHAVIORAL]: 'bg-purple-100 text-purple-700 border-purple-200',
  [NoteCategory.PREFERENCES]: 'bg-blue-100 text-blue-700 border-blue-200',
  [NoteCategory.ROUTINE]: 'bg-green-100 text-green-700 border-green-200',
  [NoteCategory.SAFETY]: 'bg-orange-100 text-orange-700 border-orange-200',
  [NoteCategory.FAMILY]: 'bg-pink-100 text-pink-700 border-pink-200',
  [NoteCategory.OTHER]: 'bg-gray-100 text-gray-700 border-gray-200',
};
